src/core/core_utils.js

```diff
@@ -28,43 +28,20 @@ function getLookupTableFactory(initializer) {
   };
 }
 
-const MissingDataException = (function MissingDataExceptionClosure() {
-  function MissingDataException(begin, end) {
+class MissingDataException extends Error {
+  constructor(begin, end) {
+    super(`Missing data [${begin}, ${end})`);
     this.begin = begin;
     this.end = end;
-    this.message = `Missing data [${begin}, ${end})`;
-  }
-
-  MissingDataException.prototype = new Error();
-  MissingDataException.prototype.name = 'MissingDataException';
-  MissingDataException.constructor = MissingDataException;
-
-  return MissingDataException;
-})();
-
-const XRefEntryException = (function XRefEntryExceptionClosure() {
-  function XRefEntryException(msg) {
-    this.message = msg;
-  }
-
-  XRefEntryException.prototype = new Error();
-  XRefEntryException.prototype.name = 'XRefEntryException';
-  XRefEntryException.constructor = XRefEntryException;
-
-  return XRefEntryException;
-})();
-
-const XRefParseException = (function XRefParseExceptionClosure() {
-  function XRefParseException(msg) {
-    this.message = msg;
   }
+}
+MissingDataException.prototype.name = 'MissingDataException';
```

Comment:
Why can't we just set the `name` property in the constructor everywhere?

Reply:
Well, that would *work* ... if we wanted to actually write the constructor in every single subclass.

Compare:
```javascript
class XRefEntryException extends Error {}
XRefEntryException.prototype.name = 'XRefEntryException';
```
with:
```javascript
class XRefEntryException extends Error {
  constructor(message) {
    super(message);
    this.name = 'XRefEntryException';
  }
}
```
I think it takes much less time to see that the two-line version is a trivial Error subclass than it does with the ... however-many-line version. And it looks like there are 7 such trivial subclasses.

My original reason for doing it that way, though, is because that is where it was in the old code **and** on the standard Error classes (see [ES6](https://www.ecma-international.org/ecma-262/6.0/#sec-nativeerror.prototype.name) and [ESNext](https://tc39.es/ecma262/#sec-nativeerror.prototype.name)). And I probably also had some silly notion that this "saves space on instances" ;-).